File: app/src/main/java/com/SE114/food_tracker/data/local/entities/Item.java
package com.SE114.food_tracker.data.local.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "item",
        foreignKeys = @ForeignKey(entity = Category.class,
                parentColumns = "category_id",
                childColumns = "category_id",
                onDelete = ForeignKey.CASCADE),
        indices = {@Index("category_id")}
)
public class Item {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "item_id")
    private int itemId;
    @ColumnInfo(name = "category_id")
    private int categoryId;
    private String name;
    @ColumnInfo(name = "time_type")
    private int timeType;
    private double price;
    private Integer rating;
    private String note;
    @ColumnInfo(name = "image_path")
    private String imagePath;
    @ColumnInfo(name = "is_synced")
    private String isSynced;
    @ColumnInfo(name = "entry_date")
    private long entryDate;
    @ColumnInfo(name = "created_at")
    private long createdAt;

    public Item() {
        this.createdAt = System.currentTimeMillis();
    }

    public int getItemId() {
        return itemId;
    }

    public void setItemId(int itemId) {
        this.itemId = itemId;
    }

    public int getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(int categoryId) {
        this.categoryId = categoryId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getTimeType() {
        return timeType;
    }

    public void setTimeType(int timeType) {
        this.timeType = timeType;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public Integer getRating() {
        return rating;
    }

    public void setRating(Integer rating) {
        this.rating = rating;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public String getImagePath() {
        return imagePath;
    }

    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }

    public String getIsSynced() {
        return isSynced;
    }

    public void setIsSynced(String isSynced) {
        this.isSynced = isSynced;
    }

    public long getEntryDate() {
        return entryDate;
    }

    public void setEntryDate(long entryDate) {
        this.entryDate = entryDate;
    }

    public long getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(long createdAt) {
        this.createdAt = createdAt;
    }
}

File: app/src/main/java/com/SE114/food_tracker/data/local/entities/Item.kt
package com.SE114.food_tracker.data.local.entities

import androidx.room.*
import java.util.UUID

@Entity(
    tableName = "item",
    foreignKeys = [ForeignKey(
        entity = Category::class,
        parentColumns = ["category_id"],
        childColumns = ["category_id"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [
        Index("category_id"),
        Index("sync_status"),
        Index("entry_date")
    ]
)
data class Item(
    @PrimaryKey
    @ColumnInfo(name = "item_id")
    var itemId: String = UUID.randomUUID().toString(), // Chuyển sang UUID String để sync không bị đè id chéo thiết bị

    @ColumnInfo(name = "category_id")
    var categoryId: String,

    var name: String,

    @ColumnInfo(name = "time_type")
    var timeType: Int,              // 0 = Sáng, 1 = Trưa/Chiều, 2 = Tối

    var price: Double,

    @ColumnInfo(name = "currency_code")
    var currencyCode: String = "VND", // Quản lý đa tiền tệ theo đặc tả

    var rating: Int? = null,
    var note: String? = null,

    @ColumnInfo(name = "image_url")
    var imageUrl: String? = null,   // Lưu path/url ảnh món ăn từ Supabase Storage

    @ColumnInfo(name = "is_shared")
    var isShared: Boolean = false,  // Chia sẻ nhật ký với bạn bè

    @ColumnInfo(name = "wallet_id")
    var walletId: String? = null,   // null = cá nhân, có giá trị = thuộc quỹ nhóm

    @ColumnInfo(name = "sync_status")
    var syncStatus: String = SyncStatus.PENDING.name,

    @ColumnInfo(name = "entry_date")
    var entryDate: Long,            // Epoch millis (UTC) của ngày ghi nhật ký

    @ColumnInfo(name = "updated_at")
    var updatedAt: Long = System.currentTimeMillis(), // Dùng so sánh Last-Write-Wins khi sync

    @ColumnInfo(name = "created_at")
    var createdAt: Long = System.currentTimeMillis()
)
